pilot/pkg/serviceregistry/kube/controller/ambientindex.go

```diff
@@ -54,29 +55,63 @@ type AmbientIndex struct {
 	byService map[string][]*model.WorkloadInfo
 	// byPod indexes by Pod IP address.
 	byPod map[string]*model.WorkloadInfo
+	// services are indexed by the network/clusterIP
+	services map[string]*workloadapi.Service
```

Comment:
nit; bump on keying this with a struct that self-documents exactly what the format for lookup is and gives compilation errors for malformed keys

Reply:
done